zk-token-sdk/src/encryption/elgamal.rs

```diff
@@ -229,29 +229,53 @@ impl ElGamalKeypair {
         &self,
         outfile: F,
     ) -> Result<String, Box<dyn std::error::Error>> {
-        let outfile = outfile.as_ref();
+        self.write_to_file(outfile)
+    }
 
-        if let Some(outdir) = outfile.parent() {
-            fs::create_dir_all(outdir)?;
-        }
+    /// Derive a keypair from an entropy seed.
+    pub fn from_seed(seed: &[u8]) -> Result<Self, Box<dyn error::Error>> {
+        let secret = ElGamalSecretKey::from_seed(seed)?;
+        let public = ElGamalPubkey::new(&secret);
+        Ok(ElGamalKeypair { public, secret })
+    }
```

Comment:
Do we need these duplicate methods with the same names as those in the trait definitions? I think they could both be implemented directly below.

Reply:
Right, I will remove these. I had `{read, write}_json` methods and I wanted to keep them for downstream, but then for methods like `from_seed` that are newly added, I guess there is no reason to have them as duplicate methods!